src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/AbstractRazorEditorTest.cs

```diff
@@ -194,16 +201,49 @@ static void WaitForFileExistsAsync(string file)
             }
         }
 
-        private static void EnsureExtensionInstalled()
+        private async Task EnsureExtensionInstalledAsync(CancellationToken cancellationToken)
         {
-            var assembly = Assembly.Load(new AssemblyName("Microsoft.VisualStudio.RazorExtension"));
-            var version = assembly.GetName().Version;
+            var assemblyName = "Microsoft.AspNetCore.Razor.LanguageServer, Version=42.42.42.42, Culture=neutral, PublicKeyToken=adb9793829ddae60";
+            using var semaphore = new SemaphoreSlim(1);
+            await semaphore.WaitAsync(cancellationToken);
+
+            AppDomain.CurrentDomain.AssemblyLoad += CurrentDomain_AssemblyLoad;
 
             var localAppData = Environment.GetEnvironmentVariable("LocalAppData");
+            Assembly? assembly = null;
+            try
+            {
+                assembly = Assembly.Load(assemblyName);
+                semaphore.Release();
+            }
+            catch (FileNotFoundException)
+            {
+                await semaphore.WaitAsync(cancellationToken);
```

Comment:
Hmm, so if we can't find our assembly we then just wait until the appdomain has loaded it? Looking at appdomain assemblies would probably enable us to prevent the catch here entirely